src/java-interop/java-interop-dlfcn.h

```diff
@@ -0,0 +1,18 @@
+#ifndef INC_JAVA_INTEROP_DLFCN_H
+#define INC_JAVA_INTEROP_DLFCN_H
+
+#include "java-interop.h"
+
+JAVA_INTEROP_BEGIN_DECLS
+
+#define JAVA_INTEROP_LIBRARY_FAILED             (-1000)
```

Comment:
If this header is to be consumed only by C++, could we use `constexpr int JAVA_INTEROP_* = VALUE;` instead? Strong typing FTW :)